sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBase.g4

```diff
@@ -790,6 +796,16 @@ ASC: 'ASC';
 DESC: 'DESC';
 FOR: 'FOR';
 INTERVAL: 'INTERVAL';
+YEAR: 'YEAR' | 'YEARS';
+MONTH: 'MONTH' | 'MONTHS';
+WEEK: 'WEEK' | 'WEEKS';
+DAY: 'DAY' | 'DAYS';
+HOUR: 'HOUR' | 'HOURS';
+MINUTE: 'MINUTE' | 'MINUTES';
+SECOND: 'SECOND' | 'SECONDS';
+MILLISECOND: 'MILLISECOND' | 'MILLISECONDS';
```

Comment:
I am wondering which systems support `MILLISECOND `, `MICROSECOND ` and `NANOSECOND `?

Reply:
nvm, it sounds like we already support them.

Reply:
yea.